src/raft.c

```diff
@@ -1667,6 +1848,9 @@ void handleRedisCommand(RedisRaftCtx *rr,RaftReq *req)
      * commands we've received this as a RAFT.ENTRY input and bundling, probably through a
      * proxy, and bundling was done before.
      */
+
+    handleAsking(req);
```

Comment:
Why do you do this here and not inside `executeRaftRedisCommandArray()` ? 

Also, why do you serialize "ASKING" differently? Why not just leave it as it is and check inside `executeRaftRedisCommandArray()` like "MULTI" here: 

https://github.com/RedisLabs/redisraft/blob/e573dd291b76591e736d74d3eea1be059aac8bad/src/raft.c#L148

Reply:
1) we have to tag asking for multis.  i.e. it will be

asking multi
get x
get y
exec

Reply:
2. the MULTI isn't visible in executeRaftRedisCommandAarray, that's in handleMulti() I'd say we are handling asking similiar to multi impractice in the manner I implemented.

Reply:
Hmm, okay I see. 

Need to take another look but if multi handling also checks for "ASKING MULTI" along with "MULTI", will it solve the problem? We can serialize "ASKING" without special serializing. Inside `executeRaftRedisCommandArray()`, as the first thing `validateRaftRedisCommandArray()` runs and takes "ASKING" out from the array. Rest of the code stays same. Something like that. 

Just saying this because it might solve the problem with less changes.

Reply:
in multi we don't serialize the multi/exec portions of it, we just serialize the array of commands in between them.  so, an asking multi would need a flag.  this seems like the cleanest way, i.e. asking is basically setting a flag on the request, and that's how we treat it.  In practice, the changes weren't that extensive (imo).

Reply:
https://github.com/RedisLabs/redisraft/blob/e573dd291b76591e736d74d3eea1be059aac8bad/src/raft.c#L148

Maybe I'm missing something but array of commands have "multi" in it, we treat it differently up above. 

I think this is not directly related with what I suggest here. So, my suggestion is keeping "asking" at the head whether it is multi or not and handling it just before we apply the command.

Here some pseudo-code, just consider if it makes sense. If it works and doesn't cause any other problems (not sure), I feel like you may avoid some of the changes, special handling (serialization, checks around executeRaftRedisCommandArray() etc). 

```c
validateRaftRedisCommandArray(commandArray) 
{
     if (commandArray->commands[0]->argv == "ASKING") {
         asking = true;
         popAskingFromArray(commandArray);
     }
     
     ...
}

executeRaftRedisCommandArray(commandArray) 
{
      if (validateRaftRedisCommandArray(commandArray) != OK) {
          return;
      }
      
      ...
      applyCommands();
      
}
```

Reply:
moved the handleAsking into RaftExecuteCommandArray() which I think addresses the issue?